kvbc/src/categorization/shared_kv_category.cpp

```diff
@@ -0,0 +1,142 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#include "categorization/shared_kv_category.h"
+
+#include "assertUtils.hpp"
+#include "categorization/column_families.h"
+#include "categorization/details.h"
+
+#include <algorithm>
+#include <map>
+#include <stdexcept>
+#include <utility>
+
+using namespace std::literals;
+
+namespace concord::kvbc::categorization::detail {
+
+const Buffer &version(BlockId block_id) {
+  static thread_local auto buf = Buffer{};
+  buf.clear();
```

Comment:
I was thinking since we have multiple places where we store those static buffers per method.
And as we spoke offline a buffer can only grow, maybe we should add a clear method that does
```
buf.clear();
if(buf.capacity() > MAX_CAP)buf.shrink_to_fit();
````